File: backend/SafetyAndHealth.Db/Models/UserProtocol.cs
using System;
namespace SafetyAndHealth.Db.Models
{
    public class UserProtocol : RelationalEntity
    {
        public string Number { get; set; }

        public DateTime StartDate { get; set; }

        public User User { get; set; }

        public string UserId { get; set; }

        public FileDescription File { get; set; }

        public long? FileId { get; set; }

        public Certificate Certificate { get; set; }

        public long CertificateId { get; set; }
    }
}

File: frontend/src/api/models/UserProtocol.ts
import { FileDescriptionDto } from "./FileDescription";
import { BaseDtoEntity } from "./BaseModel";
import { CertificateDto } from "./Certificate";

export interface UserProtocolDto extends BaseDtoEntity {
    number: string;

    startDate: Date;

    userId: string;

    file: FileDescriptionDto;

    certificate: CertificateDto;

    expiresDate?: Date;
}

export interface UserProtocolCreateUpdateDto {
    number: string;

    startDate: Date | string;

    userId: string;

    certificateId: number;
}